feature/home/src/main/java/com/epmedu/animeal/home/presentation/ui/MapboxMap.kt

```diff
@@ -68,3 +74,48 @@ private fun mapView(mapboxPublicKey: String, mapBoxStyleUri: String): MapView {
 
     return mapView
 }
+
+/**
+ * Creates a list of markers in the map
+ *
+ * @param mapView the view where the markers are being created
+ * @param feedingPoints the list of feeding points
+ * @param onFeedingPointClickListener marker click listener
+ *
+ * @return returns the manager that will allow us to update or delete the markers created
+ */
+private fun addMarkers(
+    mapView: MapView,
+    feedingPoints: List<FeedingPointUi>,
+    onFeedingPointClickListener: OnFeedingPointClickListener
+): PointAnnotationManager {
+    val annotationApi = mapView.annotations
+    val pointAnnotationManager = annotationApi.createPointAnnotationManager()
+    val markerCache = MarkerCache(mapView.context)
```

Comment:
How many times we create `MarkerCache`? 
Need to check whats going on after updating points

I expect that `MarkerCache` will recreate every time and this is a problem.

Reply:
Right now it only instantiates once, cause it is in factory parameter in `AndroidView` in `MapboxMap`. Maybe we can pass the `MarkerCache` as a parameter and create it like `mapBoxView`:

```
@Composable
fun MapboxMap(
    state: HomeState,
    onFeedingPointClick: (point: FeedingPointUi) -> Unit
) {
    val mapBoxView = mapView(state.mapBoxPublicKey, state.mapBoxStyleUri)
    val markerCache = MarkerCache(context)

    var initialLocationReceived by rememberSaveable { mutableStateOf(false) }
    LaunchedEffect(Unit) { initialLocationReceived = false }

    AndroidView(
        factory = {
            addMarkers(mapBoxView, state.feedingPoints, onFeedingPointClick, markerCache)
            mapBoxView
        },
        modifier = Modifier.fillMaxSize(),
    ) { mapView ->
        mapView.scalebar.enabled = false
        mapView.location.updateSettings {
            enabled = true
            pulsingEnabled = true
        }
```

What do you think?

Reply:
we can try, but with `remember` block